Fix issue with get_url not accepting all parameters for file module… Fixes #25263

JimCircadian opened the pull request:
Fixes (#25263) 
_Appreciate this might not be the best solution but it did what I needed from it! Will be keen to see your feedback when I'm back on Friday, thanks!!!_

##### SUMMARY
Addition to allow get_url to process state='directory' to create directory trees prior to deploying the file downloaded

##### ISSUE TYPE
 - Docs Pull Request (I guess, it fixes the fact that get_url states ALL options from file should be available!!!)

##### COMPONENT NAME
get_url / FILE_COMMON_ARGUMENTS

##### ANSIBLE VERSION
```
ansible 2.4.0 (devel 45a9f96774) last updated 2017/06/01 17:08:32 (GMT +100)
```


##### ADDITIONAL INFORMATION
Output after change:
```
TASK [Download] ***************************************************************************************************************************************************************
task path: /home/circadian/code/circadian/scratch/gist.yml:4
Using module_utils file /home/circadian/.virtualenvs/ansible-bas/src/ansible/lib/ansible/module_utils/basic.py
Using module_utils file /home/circadian/.virtualenvs/ansible-bas/src/ansible/lib/ansible/module_utils/urls.py
Using module_utils file /home/circadian/.virtualenvs/ansible-bas/src/ansible/lib/ansible/module_utils/six/__init__.py
Using module_utils file /home/circadian/.virtualenvs/ansible-bas/src/ansible/lib/ansible/module_utils/_text.py
Using module_utils file /home/circadian/.virtualenvs/ansible-bas/src/ansible/lib/ansible/module_utils/pycompat24.py
Using module_utils file /home/circadian/.virtualenvs/ansible-bas/src/ansible/lib/ansible/module_utils/six/_six.py
Using module file /home/circadian/.virtualenvs/ansible-bas/src/ansible/lib/ansible/modules/net_tools/basics/get_url.py
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: circadian
<127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/circadian/.ansible/tmp/ansible-tmp-1496335707.23-144998556057262 `" && echo ansible-tmp-1496335707.23-144998556057262="` echo /home/circadian/.ansible/tmp/ansible-tmp-1496335707.23-144998556057262 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmpfepJEy TO /home/circadian/.ansible/tmp/ansible-tmp-1496335707.23-144998556057262/get_url.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/circadian/.ansible/tmp/ansible-tmp-1496335707.23-144998556057262/ /home/circadian/.ansible/tmp/ansible-tmp-1496335707.23-144998556057262/get_url.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/home/circadian/.virtualenvs/ansible-bas/bin/python /home/circadian/.ansible/tmp/ansible-tmp-1496335707.23-144998556057262/get_url.py; rm -rf "/home/circadian/.ansible/tmp/ansible-tmp-1496335707.23-144998556057262/" > /dev/null 2>&1 && sleep 0'
changed: [localhost] => {
    "changed": true, 
    "checksum_dest": null, 
    "checksum_src": "c47883762e31eb6d88c17e8f6605efdebebc3be7", 
    "dest": "./newdirectory/plain", 
    "failed": false, 
    "gid": 1000, 
    "group": "circadian", 
    "invocation": {
        "module_args": {
            "attributes": null, 
            "backup": null, 
            "checksum": "", 
            "client_cert": null, 
            "client_key": null, 
            "content": null, 
            "delimiter": null, 
            "dest": "./newdirectory/", 
            "directory_mode": null, 
            "follow": false, 
            "force": false, 
            "force_basic_auth": false, 
            "group": null, 
            "headers": null, 
            "http_agent": "ansible-httpget", 
            "mode": null, 
            "owner": null, 
            "path": "./newdirectory/plain", 
            "regexp": null, 
            "remote_src": null, 
            "selevel": null, 
            "serole": null, 
            "setype": null, 
            "seuser": null, 
            "sha256sum": "", 
            "src": null, 
            "state": "directory", 
            "timeout": 10, 
            "tmp_dest": null, 
            "unsafe_writes": null, 
            "url": "http://ipecho.net/plain", 
            "url_password": null, 
            "url_username": null, 
            "use_proxy": true, 
            "validate_certs": true
        }
    }, 
    "md5sum": "eefa366359837a8d7d91af3c8e8e434a", 
    "mode": "0664", 
    "msg": "OK (unknown bytes)", 
    "owner": "circadian", 
    "size": 12, 
    "src": "/tmp/tmpiaedYM", 
    "state": "file", 
    "status_code": 200, 
    "uid": 1000, 
    "url": "http://ipecho.net/plain"
}
META: ran handlers
META: ran handlers
```

mechaxl committed:
Fixing policy sorting for Python 3

JimCircadian committed:
Fix issue with get_url not accepting all parameters for file module (#25263)

evancarter-iex committed:
gce_instance_template: Fix subnetwork not working (#25231)

* Fix problem with subnetwork not working

It is broken in the current master branch. This fixes it.

* Fix missing version_added

mechaxl committed:
Convert basestring to string_types (#25228)

bcoca commented:
-1 this should not be in common args as it is different per module and doesn't even make sense in some, it should be set internally by the action plugins if needed

ColOfAbRiX committed:
Fixes parted i18n issues (#24991)

* Fixed issue 24969

* Used C locale and global env update.

caphrim007 committed:
Fixes documentation (#25273)

This plugin was released with 2.4 but mentions it supports 2.3.
This patch fixes that.

jameslabocki committed:
Fix indentation for register module in example (#25274)

trondhindenes committed:
New module: win dsc (#24872)

* Added win_dsc module file

* mute output and track reboot requirements

* added tests

* proper conditionals for test

* Added moar conditionals for test

* ci fixes

* Added metadata

* fixed integration test yaml

* ci fix

* ci fix

* added module_version param and output, no longer chokes on multiple versions found.

* ci fix

* code review improvements, make return vars more pythonic, cleanup
removed reference to handles in commit message

* Fixed tests, clearer documentation

* fixed trailing whitespace

Nosmoht committed:
ipa_role: Add support for privileges (#21149)

* ipa_role: Add support for privileges

* ipa_role: Add default vaule to module documentation

mavit committed:
Support check mode in pkg5 module. (#23008)

EvanK committed:
Cron - Explicitly fail on Solaris when given a special_time (#24898)

* Added explicit failure on Solaris given a special_time

Fixes ansible/ansible#22145

* Using get_platform from module utils

n0ts committed:
support host parameter to datadog_event module (#20914)

s-hertel committed:
elasticache: add requirements to parameter_group and snapshot (#24904)

* Add requirements to elasticache parameter group

* add requirements for elasticache_snapshot

* pep8 elasticache_snapshot

* remove elasticache_snapshot from legacy files

resmo committed:
adjust versions added

mavit committed:
pkg5 with `latest` shouldn't ignore absent packages (#23006)

When the state is set to `latest` we should install absent packages, not just upgrade already-instaled packages.

This should fix #22823.

dagwieers committed:
win_timezone: Add RETURN information (#24571)

And cosmetic cleanup.

This fixes ansible/ansible-modules-extras#1767

ganeshrn committed:
Add support to fetch old style junos facts (#25250)

* Add support to fetch old style junos facts

Fixes #25050

Add support to fetch old style facts supported in Ansible <= 2.2
To fetch old style facts value `gather_subset` value shoule be `all`
and `junos-eznc` is required to be installed on control node as a
prerequisite.

* Remove unwanted import

dagwieers committed:
patch: PEP8, imports, cosmetics (#24649)

- Make PEP8 compliant
- Ensure imports are specific
- Few cosmetic changes (sort lists, casing, punctuation)

dagwieers committed:
find: PEP8, imports, cosmetics (#24646)

- Make PEP8 compliant
- Ensure imports are specific
- Few cosmetic changes (sort lists, casing, punctuation)

JimCircadian commented:
@bcoca Thanks for the response, that makes a lot of sense. I'll sort out the commit on this pull request.

JimCircadian committed:
Fix issue with get_url not accepting all parameters for file module (#25263)

JimCircadian commented:
Messed up this pull request by pulling from the upstream, apologies. Will raise again
